source/plugin/basics/data/datamanipulator.rst

```diff
@@ -0,0 +1,132 @@
+
+**Example: Getting a builder for ``WetData`` manipulators**
+
+.. code-block:: java
+
+    import org.spongepowered.api.data.DataManipulatorBuilder;
+    import org.spongepowered.api.data.DataManipulatorRegistry;
+    import org.spongepowered.api.data.manipulator.block.TreeData;
+
+    DataManipulatorRegistry registry = game.getRegistry().getManipulatorRegistry();
+    Optional<DataManipulatorBuilder<TreeData>> builder = registry.getBuilder(TreeData.class);
+
+**Example: Creating ``TreeData`` objects using the builder**
+
+.. code-block:: java
+
+    import org.spongepowered.api.data.DataHolder;
+    import org.spongepowered.api.data.DataManipulatorBuilder;
+    import org.spongepowered.api.data.manipulator.block.TreeData;
+
+    TreeData defaultData = builder.get().create();
+
+    DataHolder holder = ...;
+    Optional<TreeData> copiedData = builder.get().createFrom(holder);
+
+Manipulating Data
+=================
+
+Boolean Data
+~~~~~~~~~~~~
+
+Some data manipulators represent a single boolean value. Those manipulators usually do not contain any methods for altering their contained data, since their mere presence on a data holder translates to ``true``, while absence signals ``false``.
+
+An example for this is ``Wetdata``. A data holder is wet, if and only if a ``WetData`` object can be created from it.
+
+**Example: Check if a DataHolder contains WetData**
+
+.. code-block:: java
+
+    DataManipulatorBuilder<WetData> builder = ...;
+    DataHolder holder = ...;
+    boolean isWet = builder.createFrom(holder).isPresent();
+
+Single Value Data
+~~~~~~~~~~~~~~~~~
+
+A ``SingleValueData`` manipulator holds a single value of a specific type. For this value, ``getValue()`` and ``setValue()`` methods of the correct type are provided.
+
+Common examples for this type of data manipulator are ``TreeData`` and ``ArtData``, both of which specify the :ref:`Type <dataapi-types>` of the ``DataHolder``.
+
+**Example: Changing TreeData from representing TreeTypes.BIRCH to TreeTypes.DARK_OAK**
+
+.. code-block:: java
+
+    TreeData tree = ...;
+    if (tree.getValue() == TreeTypes.BIRCH) {
+        tree.setValue(TreeTypes.DARK_OAK);
+    }
+
+Integer Data
+~~~~~~~~~~~~
+
+``IntData`` is basically a single value data manipulator for ``Integer`` values. In addition to the above mentioned getters and setters, it has a range of valid inputs. This range is limited by a minimal and a maximal value, which are acquired by the ``getMinValue()`` and ``getMaxValue()`` methods respectively. Those boundaries are *inclusive*.
```

Comment:
Nice mentioning that they are inclusive, maybe specify an example of what inclusive means?